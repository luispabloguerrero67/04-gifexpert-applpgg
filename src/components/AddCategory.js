import React, { useState } from 'react';
import PropTypes from 'prop-types';
import TextField from '@mui/material/TextField';
import Button from '@mui/material/Button';
import Grid from '@mui/material/Grid';
import Box from '@mui/material/Box';

function AddCategory({ onchangeAddcategories, onchangeItem }) {

    const [inputvalue, setInputvalue] = useState('')

    const handleInputChange = (e) => {
        setInputvalue(e.target.value);
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputvalue.trim().length > 2) {

            onchangeAddcategories(inputvalue);
        }
    }
    const handleReset = () => {
        onchangeAddcategories('');
    }

    const handleLoadMore = () => {
        onchangeItem(10);
    }
    return (
        <>
            <Box
                sx={{
                    display: 'flex',
                    alignItems: 'center',
                    '& > :not(style)': { m: 1 },
                }}
            >
                <Grid container rowSpacing={1} columnSpacing={{ xs: 1, sm: 2, md: 3 }}>
                    <Grid item xs={12} md={4}>
                        <input placeholder='Search a gif'
                            onChange={(e) => handleInputChange(e)} type='text' 
                            className='input'/>
                        <p>{inputvalue}</p>
                        {/* 
                        <TextField
                            placeholder='Search a gif'
                            InputProps={{ style: { color: 'white' } }}
                            className='input'
                            type='text'
                            value={inputvalue}
                            onChange={(e) => handleInputChange(e)}
                        /> */}
                    </Grid>
                    <Grid item xs={2} md={2}>
                        <Button variant="contained" onClick={handleSubmit}>Search</Button>
                    </Grid>
                    <Grid item xs={2} md={2}>
                        <Button variant="contained" onClick={handleReset}>Clear</Button>
                    </Grid>

                    <Grid item xs={2} md={2}>
                        <Button variant="contained" onClick={handleLoadMore}>load more</Button>
                    </Grid>
                </Grid>
            </Box>
        </>

    )
}

AddCategory.propTypes = {
    onchangeAddcategories: PropTypes.func.isRequired,
    onchangeItem: PropTypes.func.isRequired,
}

export default AddCategory
