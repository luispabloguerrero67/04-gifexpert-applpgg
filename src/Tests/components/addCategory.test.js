import { shallow } from 'enzyme';
import React from 'react';
import AddCategory from '../../components/AddCategory';

describe('pruebas en <addcategory>', () => {

    const setCategories = jest.fn();
    const onChangeItem = jest.fn();
    const wrapper = shallow(<AddCategory onchangeAddcategories={setCategories} onchangeItem={onChangeItem} />);


    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('change on textbox', () => {

        const input = wrapper.find('input');
        const value = "Hola mundo";
        input.simulate("change", { target: { value } });
        expect(wrapper.find('p').text().trim()).toBe(value);   
    });

})