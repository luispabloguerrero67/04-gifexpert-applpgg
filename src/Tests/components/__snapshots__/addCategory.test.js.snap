// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <addcategory> debe mostrarse correctamente 1`] = `
<Fragment>
  <ForwardRef(Box)
    sx={
      Object {
        "& > :not(style)": Object {
          "m": 1,
        },
        "alignItems": "center",
        "display": "flex",
      }
    }
  >
    <ForwardRef(Grid)
      columnSpacing={
        Object {
          "md": 3,
          "sm": 2,
          "xs": 1,
        }
      }
      container={true}
      rowSpacing={1}
    >
      <ForwardRef(Grid)
        item={true}
        md={4}
        xs={12}
      >
        <input
          className="input"
          onChange={[Function]}
          placeholder="Search a gif"
          type="text"
        />
        <p />
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
        md={2}
        xs={2}
      >
        <ForwardRef(Button)
          onClick={[Function]}
          variant="contained"
        >
          Search
        </ForwardRef(Button)>
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
        md={2}
        xs={2}
      >
        <ForwardRef(Button)
          onClick={[Function]}
          variant="contained"
        >
          Clear
        </ForwardRef(Button)>
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
        md={2}
        xs={2}
      >
        <ForwardRef(Button)
          onClick={[Function]}
          variant="contained"
        >
          load more
        </ForwardRef(Button)>
      </ForwardRef(Grid)>
    </ForwardRef(Grid)>
  </ForwardRef(Box)>
</Fragment>
`;
